### PR TITLE
feat(crypto): Update webpack config to use vendored crypto relier

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We want to use the vendor crypto relier library now

## This pull request

- Updates content server and settings server config to use the vendored version

## Issue that this pull request solves

Closes: https://mozilla-hub.atlassian.net/browse/FXA-9768

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

Note, I updated this to use `node-jose` because `jose` was complaining that an `alg` was not specified. I tried to fix the underlying issue but it was turning into a rabbit hole, will file a follow up issue to use `jose` library.

### CODE REVIEW

File: packages/fxa-content-server/webpack.config.js [hunk 1]
```diff
@@ -62,8 +62,9 @@ const webpackConfig = {
       draggable: require.resolve('jquery-ui/ui/widgets/draggable'),
       duration: require.resolve('duration-js/duration'),
       'fast-text-encoding': require.resolve('fast-text-encoding'),
-      fxaCryptoDeriver: require.resolve(
-        'fxa-crypto-relier/dist/fxa-crypto-relier/fxa-crypto-deriver'
+      fxaCryptoDeriver: path.resolve(
+        __dirname,
+        '../../dist/libs/vendored/crypto-relier/src/index.js'
```
Comment (vbudhram) [inline packages/fxa-content-server/webpack.config.js:67]: I tried using `require.resolve('@fxa/vendored/crypto-relier')` would work but it didn't. Maybe I'm missing something?

Comment (chenba) [reply packages/fxa-content-server/webpack.config.js:67]: `@fxa` is not an alias that the node module resolver would know about unless there's some magic I'm not aware of.

File: packages/fxa-settings/config/webpack.config.js [hunk 2]
```diff
@@ -355,6 +356,9 @@ module.exports = function (webpackEnv) {
       fallback: {
         fs: false,
         path: false,
+        buffer: require.resolve('buffer/'),
+        crypto: require.resolve('crypto-browserify'),
+        stream: require.resolve('stream-browserify'),
```
Comment (chenba) [inline packages/fxa-settings/config/webpack.config.js:361]: Is this related to crypto-relier?

**Review by chenba — APPROVED** [review]

File: packages/fxa-content-server/webpack.config.js (hunk 1 above)
Comment (bbangert) [reply packages/fxa-content-server/webpack.config.js:67]: There's magic that I used in 123done, not sure if it'd work here but there's alias packages for various use-cases that make it work, e.g. I had to set this up:
https://github.com/mozilla/fxa/commit/9a649c6441a33823c9237956582b2bf08ac23cc9#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R265

And to get 123done to use it:
https://github.com/mozilla/fxa/commit/9a649c6441a33823c9237956582b2bf08ac23cc9#diff-82cb871fa708015fd2a5e89be112959108a1839c19d3cc28b7560f2a38f911aeR1

I'm not sure if module-alias works for webpack, but perhaps there's something similar if it doesn't?

File: packages/fxa-settings/config/webpack.config.js (hunk 2 above)
Comment (bbangert) [reply packages/fxa-settings/config/webpack.config.js:361]: I presume its because fxa-crypto-relier was pulling these in, and now webpack needs a direct reference itself?

File: libs/vendored/crypto-relier/src/lib/deriver/deriver-utils.ts [hunk 3]
```diff
@@ -16,29 +16,43 @@ export class DeriverUtils {
    * @param {string} bundle - String bundle to encrypt using the provided key
    * @returns {Promise}
    */
-  async encryptBundle(appPublicKeyJwk: string, bundle: string) {
-    const rawKey = jose.decodeJwt('.' + appPublicKeyJwk + '.');
-    const key = await jose.importJWK(rawKey);
+  encryptBundle(appPublicKeyJwk: string, bundle: string) {
```
Comment (vbudhram) [inline libs/vendored/crypto-relier/src/lib/deriver/deriver-utils.ts:19]: This is same as https://github.com/mozilla/fxa-crypto-relier/blob/c454131c4befaf007e6f245b1673f5078a34c5c5/src/deriver/DeriverUtils.js

File: libs/vendored/crypto-relier/src/lib/deriver/deriver-utils.ts [hunk 4]
```diff
@@ -16,29 +16,43 @@ export class DeriverUtils {
...
-      throw new Error('appJwk is not on curve P-256');
-    }
-    if ('d' in rawKey) {
-      throw new Error('appJwk includes the private key');
-    }
+    return jose.JWK.asKey(appJwk).then(function (key: {
+      kty: string;
+      get: (arg0: string) => string;
+      has: (arg0: string, arg1: boolean) => any;
+    }) {
```
Comment (bbangert) [inline libs/vendored/crypto-relier/src/lib/deriver/deriver-utils.ts:27]: We could undo the promise here to retain async/await.

File: package.json [hunk 5]
```diff
@@ -103,7 +103,7 @@
     "next-auth": "beta",
     "node-fetch": "^2.6.7",
     "node-hkdf": "^0.0.2",
-    "node-jose": "^2.2.0",
+    "node-jose": "1.0.0",
```
Comment (bbangert) [inline package.json:106]: The latest version doesn't work?

**Review by bbangert — APPROVED** [review]

**Comment by darktrojan** [comment]
This appears to work, thanks so much! :+1:
